### PR TITLE
MHV-64147: Blue Button reducer implemented

### PR DESCRIPTION
**Pull request by mattwrightva**
## Summary
Created a reducer for the following Blue Button data:
- medications
- appointments
- demographics
- military service
- account summary (treatment facilities)

Added mock data endpoints for all of the above data

## Related issue(s)
https://jira.devops.va.gov/browse/MHV-64147

> ### Create reducer for remaining BB download domains
> Description
> Work w/ Mike for data mapping questions.
> 
> This includes ALL data that we aren't currently pulling for BB download, including medications, appts, demographics, etc.

## Testing done
Unit tests have been written for the additions to MrApi and the new Blue Button reducer. 

## Screenshots
No UI changes

## What areas of the site does it impact?
MHV medical records

## Acceptance criteria
Blue Button reducer implemented, working, and formatting data for Blue Button report

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/mhv-medical-records/reducers/blueButton.js [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+      street: `${item.permStreet1} ${item.permStreet2 || ''}`.trim(),
+      city: item.permCity,
+      state: item.permState,
+      zipcode: item.permZipcode,
+      country: item.permCountry,
+    },
+    contactInfo: {
+      homePhone: 'None recorded',
+      workPhone: 'None recorded',
+      cellPhone: 'None recorded',
```
Comment (mmoyer-va) [inline src/applications/mhv-medical-records/reducers/blueButton.js:147]: Are we assuming no data provided for these values as well?

File: src/applications/mhv-medical-records/reducers/blueButton.js [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+      : {};
+
+  return {
+    authenticationSummary: authenticationInfo,
+    vaTreatmentFacilities: mappedFacilities,
+  };
+};
+
+export const blueButtonReducer = (state = initialState, action) => {
+  if (action.type === Actions.BlueButtonReport.GET) {
```
Comment (mmoyer-va) [inline src/applications/mhv-medical-records/reducers/blueButton.js:251]: For consistency, use the `switch` statement and just disable the linting for that line.

**Review by mmoyer-va — CHANGES_REQUESTED** [review]
Two overall comments:
1. Write some tests for these conversion functions.
2. Breaking some of the bigger ones (e.g. demographics) into smaller functions might make it easier to test some functionality, but I'll leave that up to you.

File: src/applications/mhv-medical-records/reducers/blueButton.js [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+   * @type {Array}
+   */
+  accountSummary: undefined,
+};
+
+/**
+ * Convert the FHIR vaccine resource from the backend into the appropriate model.
+ * @param {Object} vaccine a FHIR vaccine resource
+ * @returns a vaccine object that this application can use, or null if the param is null/undefined
+ */
```
[the remark this answers is not in the document]
Comment (mattwrightva) [reply src/applications/mhv-medical-records/reducers/blueButton.js]: I updated these docs and removed the empty line spacing. Good catch!

File: src/applications/mhv-medical-records/reducers/blueButton.js [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+          }, ${location.physicalAddress.state} ${
+            location.physicalAddress.postalCode
+          }`
+        : 'No address available',
+      clinicName: attributes.clinic || 'Unknown Clinic',
+      location: clinic.physicalLocation || 'Unknown Location',
+      clinicPhone: clinic.phoneNumber || 'N/A',
+    },
+    detailsShared: {
+      reason: attributes.serviceCategory?.[0]?.text || 'Not specified',
```
[the remark this answers is not in the document]
Comment (mattwrightva) [reply src/applications/mhv-medical-records/reducers/blueButton.js]: Switched this to allow multiple. I will double check with UCD about this. Perhaps we should look at legacy also.

File: src/applications/mhv-medical-records/reducers/blueButton.js [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+    middleName: item.middleName || 'None recorded',
+    lastName: item.lastName,
+    dateOfBirth: new Date(item.dateOfBirth).toLocaleDateString('en-US', {
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+    }),
+    age: item.age,
+    gender: item.gender,
+    ethnicity: 'None recorded', // Assuming no data provided
```
[the remark this answers is not in the document]
Comment (mattwrightva) [reply src/applications/mhv-medical-records/reducers/blueButton.js]: Done

File: src/applications/mhv-medical-records/reducers/blueButton.js [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+};
+
+/**
+ * Convert the FHIR vaccine resource from the backend into the appropriate model.
+ * @param {Object} vaccine a FHIR vaccine resource
+ * @returns a vaccine object that this application can use, or null if the param is null/undefined
+ */
+
+export const convertMedication = med => {
+  if (typeof med === 'undefined' || med === null) {
```
[the remark this answers is not in the document]
Comment (mattwrightva) [reply src/applications/mhv-medical-records/reducers/blueButton.js:37]: Good point! Done.

File: src/applications/mhv-medical-records/reducers/blueButton.js [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+    what: attributes.serviceName || 'General',
+    where: {
+      facilityName: location.name || 'Unknown Facility',
+      address: location.physicalAddress
+        ? `${location.physicalAddress.line.join(' ')}, ${
+            location.physicalAddress.city
+          }, ${location.physicalAddress.state} ${
+            location.physicalAddress.postalCode
+          }`
+        : 'No address available',
```
[the remark this answers is not in the document]
Comment (mattwrightva) [reply src/applications/mhv-medical-records/reducers/blueButton.js:95]: Mine are actually meant to be on separate lines anyway, so I'm just going to keep it as the original physicalAddress object and deal with the line renderings when creating the PDF.

File: src/applications/mhv-medical-records/reducers/blueButton.js (hunk 2 above)
Comment (mattwrightva) [reply src/applications/mhv-medical-records/reducers/blueButton.js]: No problem

**Review by mmoyer-va — APPROVED** [review]
Changes look good.

**Review by rsmithadhoc — APPROVED** [review]

**Pull request merged**